test/dialing/direct.node.ts

```diff
@@ -103,6 +106,19 @@ describe('Dialing (direct, TCP)', () => {
     await connection.close()
   })
 
+  it('should be able to connect to remote node with duplicated addresses', async () => {
```

Comment:
Please can an assertion be added here to ensure there are multiple addresses in play?

Reply:
Not sure I understand. I've set `maxAddrsToDial` to 1 so without filtering duplicates, this test fails as it tries to dial with 2 duplicated addresses. 

If you meant to assert there were duplicates before filtering, I don't think it's possible without code instrumentation as it happens inside the pipe: https://github.com/libp2p/js-libp2p/blob/master/src/connection-manager/dialer/index.ts#L238 (one address comes from dial param and other from resolving address in peer book).

This bug doesn't happen if address book contains resolved duplicate^^